app/shop.py

```diff
@@ -0,0 +1,50 @@
+class Shop:
+    def __init__(self, name: str, price_list: dict, location: list) -> None:
+        self.name = name
+        self.price_list = price_list
+        self.location = location
+
+    def calculate_purchase_cost(self, products_to_buy: dict) -> float:
+        cost = 0
+
+        for key in products_to_buy.keys():
+            cost += round(self.price_list[key] * products_to_buy[key], 2)
+
+        return cost
```

Comment:
try to use `sum` and comprehensions

Reply:
Fixed. Pretty elegant solution. Thanks. In JS I would use reduce method)))